lib/src/realm_class.dart

```diff
@@ -496,8 +496,13 @@ class Realm implements Finalizable {
   }
 
   static set logger(Logger value) {
+    if (_logger == value) {
+      return;
+    }
+
     _logger.clearListeners();
-    _logger = (value is _RealmLogger) ? value : _RealmLogger(value);
+    _logger = value;
+    _logger.onLevelChanged.listen((logLevel) => realmCore.loggerSetLogLevel(logLevel ?? RealmLogLevel.off, scheduler.nativePort));
```

Comment:
Call `realmCore.loggerSetLogLevel` before attaching to the listener for the level.

Reply:
fixing